### PR TITLE
Closes CPPL-70

### PR DESCRIPTION
**Pull request by iodream**
Update Income functionality

### CODE REVIEW

File: src/Client/Controller/LoginPageController.cpp [hunk 1]
```diff
@@ -30,6 +33,11 @@ void LoginPageController::ConnectPage()
 		&LoginPage::GotoSignup,
 		this,
 		&LoginPageController::OnGoToSignupPage);
+
+#ifdef QT_DEBUG
+	QShortcut *shortcut = new QShortcut(QKeySequence("Return"), &m_page);
+	connect(shortcut, SIGNAL(activated()), this, SLOT(QuickLogin()));
```
Comment (yaroslav-kernitskyi) [inline src/Client/Controller/LoginPageController.cpp:39]: Consider using modern signal-slot syntax

File: src/Client/Controller/LoginPageController.cpp [hunk 2]
```diff
@@ -75,3 +83,13 @@ void LoginPageController::OnGoToSignupPage()
 {
 	emit ChangePage(UIPages::SIGNUP);
 }
+
+#ifdef QT_DEBUG
+void LoginPageController::QuickLogin()
```
Comment (yaroslav-kernitskyi) [inline src/Client/Controller/LoginPageController.cpp:88]: I don't really like this, i know it is for convenience but you should probably keep such thing in local repo. Databases of other team members can be completely different and this thing just will not work.

Comment (iodream) [reply src/Client/Controller/LoginPageController.cpp:88]: i could remove it but i thought this was the standard user-pass we were all using for debugging. i can also replace it with a constant variable. 

in my experience it has sped up client debugging a lot.

File: src/Client/Controller/MainPage/ListPagesController.cpp [hunk 3]
```diff
@@ -183,7 +183,7 @@ void ListPagesController::OnCreateList()
 
 	List new_list;
 	new_list.name = m_create_page.GetListName();
-	new_list.state.id = 11 + m_create_page.GetListState();
+	new_list.state.id = 1 + m_create_page.GetListState();
```
Comment (yaroslav-kernitskyi) [inline src/Client/Controller/MainPage/ListPagesController.cpp:186]: I know it's not you wrote that code, but `GetListState` calls `m_ui->ListState->currentIndex()` is not guaranteed to return valid list state id. Probably `m_ui->ListState->currentData()` should be used.

File: src/Client/Utils.cpp [hunk 4]
```diff
@@ -1,4 +1,11 @@
-#include "Utils.h"
+ #include "Utils.h"
+
+std::string EraseWhitespace(std::string str)
+{
+	str.erase(std::remove_if(str.begin(), str.end(), ::isspace)
+			, str.end());
```
Comment (yaroslav-kernitskyi) [inline src/Client/Utils.cpp:6]: Is this ok with our code convention?

File: src/Client/View/MainPage/Income/IncomeEditSubPage/IncomeEditSubPage.cpp [hunk 5]
```diff
@@ -1,13 +1,45 @@
...
+		&IncomeEditSubPage::OnEditIncome);
+}
+
+
+void IncomeEditSubPage::OnEditIncome()
+{
+	Income updated_income;
+	updated_income.add_time = QDate::currentDate().toString();
+	updated_income.amount = ui->AmountLineEdit->text().toLongLong();
+	updated_income.category.id = 1;
```
Comment (yaroslav-kernitskyi) [inline src/Client/View/MainPage/Income/IncomeEditSubPage/IncomeEditSubPage.cpp:24]: Will this be fixed in current sprint?

File: src/Client/Controller/LoginPageController.cpp (hunk 2 above)
Comment (yaroslav-kernitskyi) [reply src/Client/Controller/LoginPageController.cpp:88]: I understand why you added the thing, but I'm not sure this should be in repo.

**Review by yaroslav-kernitskyi — APPROVED** [review]

File: src/Client/Controller/MainPage/ListPagesController.cpp (hunk 3 above)
Comment (iodream) [reply src/Client/Controller/MainPage/ListPagesController.cpp:186]: so how should i assign it then? currentData() doesnt return an IdType

File: src/Client/Utils.cpp (hunk 4 above)
Comment (iodream) [reply src/Client/Utils.cpp:6]: im not sure but ive changed it a bit

**Review by MaksymTerekhovskyy — APPROVED** [review]

**Pull request merged**